src/random.h

```diff
@@ -25,6 +26,29 @@ int GetRandInt(int nMax);
 uint256 GetRandHash();
 
 /**
+ * Rearranges the elements in the range [first,first+len) randomly, assuming
+ * that gen is a uniform random number generator. Follows the same algorithm as
+ * std::shuffle in C++11 (a Durstenfeld shuffle).
```

Comment:
@bitcartel originally used that approach, but the specification of ``std::shuffle`` and ``URNG`` are not strong enough to guarantee determinism (needed for ``first`` and ``firstMap`` to be permuted in the same way), use of a secure RNG, and a uniform distribution over permutations.